ironfish/src/network/peers/peer.ts

```diff
@@ -289,14 +289,25 @@ export class Peer {
       }
     }
 
-    // The remaining states are CONNECTING
-    return {
-      type: 'CONNECTING',
-      identity: this.state.identity,
-      connections: {
-        ...(webRtcConnection ? { webRtc: webRtcConnection } : {}),
-        ...(wsConnection ? { webSocket: wsConnection } : {}),
-      },
+    if (webRtcConnection) {
+      return {
+        type: 'CONNECTING',
+        identity: this.state.identity,
+        connections: {
+          webRtc: webRtcConnection,
+          webSocket: wsConnection,
+        },
+      }
+    } else if (wsConnection) {
+      return {
+        type: 'CONNECTING',
+        identity: this.state.identity,
+        connections: {
+          webSocket: wsConnection,
```

Comment:
This should carry forward the `webRtcConnection` as well -- once that's updated, this looks good.

I'm not sure if there was a regression in TS here, or if I'm missing something, or it's not capable of narrowing types in these cases. I'd expect the following two code blocks to be equivalent, but the one with the `OR` operator has a TS error:

1.
```js
if (webRtcConnection) {
    return {
        type: 'CONNECTING',
        identity: this.state.identity,
        connections: {
            webRtc: webRtcConnection,
            webSocket: wsConnection,
        },
    }
}

if (wsConnection) {
    return {
        type: 'CONNECTING',
        identity: this.state.identity,
        connections: {
            webRtc: webRtcConnection,
            webSocket: wsConnection,
        },
    }
}
```

2.
```js
if (webRtcConnection || wsConnection) {
    return {
        type: 'CONNECTING',
        identity: this.state.identity,
        connections: {
            webRtc: webRtcConnection,
            webSocket: wsConnection,
        },
    }
}
```